ext/rugged/rugged_patch.c

```diff
@@ -355,14 +372,35 @@ static int patch_print_cb(
 static VALUE rb_git_diff_patch_to_s(VALUE self)
 {
 	git_patch *patch;
-	VALUE rb_str = rb_str_new(NULL, 0);
+	VALUE rb_buffer = rb_ary_new();
+	Data_Get_Struct(self, git_patch, patch);
+
+	rugged_exception_check(git_patch_print(patch, patch_print_cb, (void*)rb_buffer));
+
+	return rb_ary_join(rb_buffer, Qnil);
```

Comment:
Could we expose `git_patch_print` to Ruby?  Then we could implement this (and the other place that uses `git_patch_print`) in Ruby.